libffi does not compile on macOS 14 with Xcode 16.2

Find logs (including config.log) below

```
(base) johnzhou@Johns-MBP ~ % git clone https://github.com/libffi/libffi.git
Cloning into 'libffi'...
remote: Enumerating objects: 15263, done.
remote: Counting objects: 100% (1360/1360), done.
remote: Compressing objects: 100% (201/201), done.
remote: Total 15263 (delta 1248), reused 1161 (delta 1159), pack-reused 13903 (from 3)
Receiving objects: 100% (15263/15263), 8.37 MiB | 22.44 MiB/s, done.
Resolving deltas: 100% (9552/9552), done.
(base) johnzhou@Johns-MBP ~ % cd libffi
(base) johnzhou@Johns-MBP libffi % autoreconf -ivf
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: glibtoolize --copy --force
glibtoolize: putting auxiliary files in '.'.
glibtoolize: copying file './ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
glibtoolize: copying file 'm4/libtool.m4'
glibtoolize: copying file 'm4/ltoptions.m4'
glibtoolize: copying file 'm4/ltsugar.m4'
glibtoolize: copying file 'm4/ltversion.m4'
glibtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /opt/homebrew/Cellar/autoconf/2.72/bin/autoconf --force
configure.ac:88: warning: The preprocessor macro `STDC_HEADERS' is obsolete.
configure.ac:88:   Except in unusual embedded environments, you can safely include all
configure.ac:88:   ISO C90 headers unconditionally.
configure.ac:123: warning: The macro 'AC_TRY_COMPILE' is obsolete.
configure.ac:123: You should run autoupdate.
./lib/autoconf/general.m4:2845: AC_TRY_COMPILE is expanded from...
lib/m4sugar/m4sh.m4:690: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:697: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/asmcfi.m4:1: GCC_AS_CFI_PSEUDO_OP is expanded from...
configure.ac:123: the top level
autoreconf: running: /opt/homebrew/Cellar/autoconf/2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:31: installing './compile'
configure.ac:19: installing './install-sh'
configure.ac:19: installing './missing'
Makefile.am: installing './depcomp'
doc/Makefile.am:3: installing 'doc/mdate-sh'
doc/Makefile.am:3: installing 'doc/texinfo.tex'
autoreconf: Leaving directory '.'
(base) johnzhou@Johns-MBP libffi % python generate-darwin-source-and-headers.py
checking build system type... aarch64-apple-darwin
checking host system type... Invalid configuration 'i386-apple-ios-simulator': Kernel 'ios' not known to work with OS 'simulator'.
configure: error: /bin/sh ../config.sub i386-apple-ios-simulator failed
Traceback (most recent call last):
  File "/Users/johnzhou/libffi/generate-darwin-source-and-headers.py", line 307, in <module>
    generate_source_and_headers(
  File "/Users/johnzhou/libffi/generate-darwin-source-and-headers.py", line 273, in generate_source_and_headers
    build_target(ios_simulator_i386_platform, platform_headers)
  File "/Users/johnzhou/libffi/generate-darwin-source-and-headers.py", line 217, in build_target
    subprocess.check_call(
  File "/opt/anaconda3/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../configure', '--host=i386-apple-ios-simulator', '--build=arm64-apple-darwin']' returned non-zero exit status 1.
(base) johnzhou@Johns-MBP libffi % sw_vers
ProductName:		macOS
ProductVersion:		14.7.4
BuildVersion:		23H420
(base) johnzhou@Johns-MBP libffi % cat build_iphonesimulator-i386/config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by libffi configure 3.4.8, which was
generated by GNU Autoconf 2.72.  Invocation command line was

  $ ../configure --host=i386-apple-ios-simulator --build=arm64-apple-darwin

## --------- ##
## Platform. ##
## --------- ##

hostname = Johns-MBP
uname -m = arm64
uname -r = 23.6.0
uname -s = Darwin
uname -v = Darwin Kernel Version 23.6.0: Thu Dec 19 20:44:10 PST 2024; root:xnu-10063.141.1.703.2~1/RELEASE_ARM64_T6000

/usr/bin/uname -p = arm
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = Mach kernel version:
	 Darwin Kernel Version 23.6.0: Thu Dec 19 20:44:10 PST 2024; root:xnu-10063.141.1.703.2~1/RELEASE_ARM64_T6000
Kernel configured for up to 10 processors.
10 processors are physically available.
10 processors are logically available.
Processor type: arm64e (ARM64E)
Processors active: 0 1 2 3 4 5 6 7 8 9
Primary memory available: 16.00 gigabytes
Default processor set: 636 tasks, 3675 threads, 10 processors
Load average: 4.63, Mach factor: 6.02
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /usr/gnu/bin/
PATH: /usr/local/bin/
PATH: /bin/
PATH: /usr/bin/
PATH: ./


## ----------- ##
## Core tests. ##
## ----------- ##

configure:3046: looking for aux files: ltmain.sh compile missing install-sh config.guess config.sub
configure:3059:  trying ../
configure:3088:   ../ltmain.sh found
configure:3088:   ../compile found
configure:3088:   ../missing found
configure:3070:   ../install-sh found
configure:3088:   ../config.guess found
configure:3088:   ../config.sub found
configure:3210: checking build system type
configure:3226: result: aarch64-apple-darwin
configure:3246: checking host system type
configure:3256: error: /bin/sh ../config.sub i386-apple-ios-simulator failed

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_build=aarch64-apple-darwin
ac_cv_env_CCASFLAGS_set=
ac_cv_env_CCASFLAGS_value=
ac_cv_env_CCAS_set=
ac_cv_env_CCAS_value=
ac_cv_env_CPPFLAGS_set=
ac_cv_env_CPPFLAGS_value=
ac_cv_env_CXXCPP_set=
ac_cv_env_CXXCPP_value=
ac_cv_env_LT_SYS_LIBRARY_PATH_set=
ac_cv_env_LT_SYS_LIBRARY_PATH_value=
ac_cv_env_build_alias_set=set
ac_cv_env_build_alias_value=arm64-apple-darwin
ac_cv_env_host_alias_set=set
ac_cv_env_host_alias_value=i386-apple-ios-simulator
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=

## ----------------- ##
## Output variables. ##
## ----------------- ##

ACLOCAL=''
AMDEPBACKSLASH=''
AMDEP_FALSE=''
AMDEP_TRUE=''
AMTAR=''
AM_BACKSLASH=''
AM_DEFAULT_V=''
AM_DEFAULT_VERBOSITY=''
AM_LTLDFLAGS=''
AM_RUNTESTFLAGS=''
AM_V=''
AR=''
AUTOCONF=''
AUTOHEADER=''
AUTOMAKE=''
AWK=''
BUILD_DOCS_FALSE=''
BUILD_DOCS_TRUE=''
CC='xcrun -sdk iphonesimulator clang -target i386-apple-ios-simulator'
CCAS=''
CCASDEPMODE=''
CCASFLAGS=''
CCDEPMODE=''
CFLAGS='-miphoneos-version-min=7.0 -fembed-bitcode'
CPPFLAGS=''
CSCOPE=''
CTAGS=''
CXX=''
CXXCPP=''
CXXDEPMODE=''
CXXFLAGS=''
CYGPATH_W=''
DEFS=''
DEPDIR=''
DLLTOOL=''
DSYMUTIL=''
DUMPBIN=''
ECHO_C='\c'
ECHO_N=''
ECHO_T=''
EGREP=''
ETAGS=''
EXEEXT=''
FFI_DEBUG_FALSE=''
FFI_DEBUG_TRUE=''
FFI_EXEC_TRAMPOLINE_TABLE=''
FFI_EXEC_TRAMPOLINE_TABLE_FALSE=''
FFI_EXEC_TRAMPOLINE_TABLE_TRUE=''
FGREP=''
FILECMD=''
GREP=''
HAVE_LONG_DOUBLE=''
HAVE_LONG_DOUBLE_VARIANT=''
INSTALL_DATA=''
INSTALL_PROGRAM=''
INSTALL_SCRIPT=''
INSTALL_STRIP_PROGRAM=''
LD='xcrun -sdk iphonesimulator ld -target i386-apple-ios-simulator'
LDFLAGS=''
LIBFFI_BUILD_VERSIONED_SHLIB_FALSE=''
LIBFFI_BUILD_VERSIONED_SHLIB_GNU_FALSE=''
LIBFFI_BUILD_VERSIONED_SHLIB_GNU_TRUE=''
LIBFFI_BUILD_VERSIONED_SHLIB_SUN_FALSE=''
LIBFFI_BUILD_VERSIONED_SHLIB_SUN_TRUE=''
LIBFFI_BUILD_VERSIONED_SHLIB_TRUE=''
LIBOBJS=''
LIBS=''
LIBTOOL=''
LIPO=''
LN_S=''
LTLIBOBJS=''
LT_SYS_LIBRARY_PATH=''
MAINT=''
MAINTAINER_MODE_FALSE=''
MAINTAINER_MODE_TRUE=''
MAKEINFO=''
MANIFEST_TOOL=''
MKDIR_P=''
NM=''
NMEDIT=''
OBJDUMP=''
OBJEXT=''
OPT_LDFLAGS=''
OTOOL64=''
OTOOL=''
PACKAGE=''
PACKAGE_BUGREPORT='http://github.com/libffi/libffi/issues'
PACKAGE_NAME='libffi'
PACKAGE_STRING='libffi 3.4.8'
PACKAGE_TARNAME='libffi'
PACKAGE_URL=''
PACKAGE_VERSION='3.4.8'
PATH_SEPARATOR=':'
PRTDIAG=''
RANLIB=''
READELF=''
SECTION_LDFLAGS=''
SED=''
SET_MAKE=''
SHELL='/bin/sh'
STRIP=''
TARGET=''
TARGETDIR=''
TARGET_OBJ=''
TESTSUBDIR_FALSE=''
TESTSUBDIR_TRUE=''
VERSION=''
ac_ct_AR=''
ac_ct_CC=''
ac_ct_CXX=''
ac_ct_DUMPBIN=''
ac_ct_READELF=''
am__EXEEXT_FALSE=''
am__EXEEXT_TRUE=''
am__fastdepCCAS_FALSE=''
am__fastdepCCAS_TRUE=''
am__fastdepCC_FALSE=''
am__fastdepCC_TRUE=''
am__fastdepCXX_FALSE=''
am__fastdepCXX_TRUE=''
am__include=''
am__isrc=''
am__leading_dot=''
am__nodep=''
am__quote=''
am__rm_f_notfound=''
am__tar=''
am__untar=''
am__xargs_n=''
ax_enable_builddir_sed=''
bindir='${exec_prefix}/bin'
build='aarch64-apple-darwin'
build_alias='arm64-apple-darwin'
build_cpu='aarch64'
build_os='darwin'
build_vendor='apple'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
docdir='${datarootdir}/doc/${PACKAGE_TARNAME}'
dvidir='${docdir}'
exec_prefix='NONE'
host='i386-apple-ios-simulator'
host_alias='i386-apple-ios-simulator'
host_cpu=''
host_os=''
host_vendor=''
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
install_sh=''
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
mkdir_p=''
oldincludedir='/usr/include'
pdfdir='${docdir}'
prefix='NONE'
program_transform_name='s,x,x,'
psdir='${docdir}'
runstatedir='${localstatedir}/run'
sbindir='${exec_prefix}/sbin'
sharedstatedir='${prefix}/com'
sys_symbol_underscore=''
sysconfdir='${prefix}/etc'
target=''
target_alias=''
target_cpu=''
target_os=''
target_vendor=''
tmake_file=''
toolexecdir=''
toolexeclibdir=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME "libffi"
#define PACKAGE_TARNAME "libffi"
#define PACKAGE_VERSION "3.4.8"
#define PACKAGE_STRING "libffi 3.4.8"
#define PACKAGE_BUGREPORT "http://github.com/libffi/libffi/issues"
#define PACKAGE_URL ""

configure: exit 1
(base) johnzhou@Johns-MBP libffi % 
```

--- a/generate-darwin-source-and-headers.py
+++ b/generate-darwin-source-and-headers.py
@@ -11,15 +11,6 @@ class Platform(object):
     pass
 
 
-class i386_platform(Platform):
-    arch = 'i386'
-
-    prefix = "#ifdef __i386__\n\n"
-    suffix = "\n\n#endif"
-    src_dir = 'x86'
-    src_files = ['sysv.S', 'ffi.c', 'internal.h']
-
-
 class x86_64_platform(Platform):
     arch = 'x86_64'
 
@@ -45,13 +36,6 @@ class armv7_platform(Platform):
     suffix = "\n\n#endif"
     src_dir = 'arm'
     src_files = ['sysv.S', 'ffi.c', 'internal.h']
-
-
-class ios_simulator_i386_platform(i386_platform):
-    target = 'i386-apple-ios-simulator'
-    directory = 'darwin_ios'
-    sdk = 'iphonesimulator'
-    version_min = '-miphoneos-version-min=7.0'
 
 
 class ios_simulator_x86_64_platform(x86_64_platform):
@@ -115,13 +99,6 @@ class tvos_device_arm64_platform(arm64_platform):
     directory = 'darwin_tvos'
     sdk = 'appletvos'
     version_min = '-mtvos-version-min=9.0'
-
-
-class watchos_simulator_i386_platform(i386_platform):
-    target = 'i386-apple-watchos-simulator'
-    directory = 'darwin_watchos'
-    sdk = 'watchsimulator'
-    version_min = '-mwatchos-version-min=4.0'
 
 
 class watchos_simulator_x86_64_platform(x86_64_platform):
@@ -210,7 +187,7 @@ def build_target(platform, platform_headers):
     mkdir_p(build_dir)
     env = dict(CC=xcrun_cmd('clang'),
                LD=xcrun_cmd('ld'),
-               CFLAGS='%s -fembed-bitcode' % (platform.version_min))
+               CFLAGS='%s' % (platform.version_min))
     working_dir = os.getcwd()
     try:
         os.chdir(build_dir)
@@ -248,7 +225,6 @@ def generate_source_and_headers(
     copy_files('include', 'darwin_common/include', pattern='*.h')
 
     if generate_ios:
-        copy_src_platform_files(ios_simulator_i386_platform)
         copy_src_platform_files(ios_simulator_x86_64_platform)
         copy_src_platform_files(ios_simulator_arm64_platform)
         copy_src_platform_files(ios_device_armv7_platform)
@@ -261,7 +237,6 @@ def generate_source_and_headers(
         copy_src_platform_files(tvos_simulator_arm64_platform)
         copy_src_platform_files(tvos_device_arm64_platform)
     if generate_watchos:
-        copy_src_platform_files(watchos_simulator_i386_platform)
         copy_src_platform_files(watchos_simulator_x86_64_platform)
         copy_src_platform_files(watchos_simulator_arm64_platform)
         copy_src_platform_files(watchos_device_armv7k_platform)
@@ -270,7 +245,6 @@ def generate_source_and_headers(
     platform_headers = collections.defaultdict(set)
 
     if generate_ios:
-        build_target(ios_simulator_i386_platform, platform_headers)
         build_target(ios_simulator_x86_64_platform, platform_headers)
         build_target(ios_simulator_arm64_platform, platform_headers)
         build_target(ios_device_armv7_platform, platform_headers)
@@ -283,7 +257,6 @@ def generate_source_and_headers(
         build_target(tvos_simulator_arm64_platform, platform_headers)
         build_target(tvos_device_arm64_platform, platform_headers)
     if generate_watchos:
-        build_target(watchos_simulator_i386_platform, platform_headers)
         build_target(watchos_simulator_x86_64_platform, platform_headers)
         build_target(watchos_simulator_arm64_platform, platform_headers)
         build_target(watchos_device_armv7k_platform, platform_headers)

